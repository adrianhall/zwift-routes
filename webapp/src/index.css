body, div#root {
  height: 100%;
  left: 0;
  margin: 0;
  padding: 0;
  position: fixed;
  top: 0;
  width: 100%;
}
