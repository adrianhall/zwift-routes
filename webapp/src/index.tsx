import React from 'react';
import ReactDOM from 'react-dom';

const Application: React.SFC<{}> = () => (
  <h1>In Application</h1>
);

ReactDOM.render(<Application />, document.getElementById('root'));
