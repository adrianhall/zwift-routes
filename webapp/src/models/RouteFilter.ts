import ZwiftSport from './ZwiftSport';

/**
 * Definition of the filters that can be applied to the 
 * route list.
 */
export default interface RouteFilter {
  /**
   * Filter by the Zwift World.  If set, matches zwiftWorld.
   */
  world?: string;

  /**
   * Filter by the Sport.  If set, matches 
   */
  sport?: ZwiftSport;

  /**
   * Include completed routes in the set.
   */
  includeCompletedRoutes: boolean;

  /**
   * Include event only routes in the set.
   */
  includeEventOnlyRoutes: boolean;
  
  /**
   * Include the default world (Watopia) in the list.
   */
  includeDefaultWorld: boolean;

  /**
   * Include if the route if above the minimum route.
   */
  maximumZwiftLevel?: number;
}
