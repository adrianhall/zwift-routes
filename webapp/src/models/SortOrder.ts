import SortField from "./SortField";

interface SortOrder {
  /**
   * What field within the Zwift Route to order by
   */
  orderBy: SortField;

  /**
   * Ascending (true) or Descening?
   */
  ascending: boolean;
}

export default SortOrder;
