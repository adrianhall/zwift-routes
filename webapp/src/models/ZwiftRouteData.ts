/**
 * Definition for the static user data that is stored in routes.json
 */
export default interface ZwiftRouteData {
  /**
   * The unique ID for the route.
   */
  id: string;

  /**
   * The VeloViewer difficulty rating
   */
  difficulty: number;

  /**
   * The route distance (in km).
   */
  distance: number;

  /**
   * The elevation gain for the route (in m).
   */
  elevation: number;

  /**
   * True if this route is available only in events.
   */
  eventonly: boolean;

  /**
   * The distance from the drop-point to the route start (in km).
   */
  leadin_distance: number;

  /**
   * The elevation gain from the drop-point to the route start (in m).
   */
  leadin_elevation: number;

  /**
   * Minimum Zwift level required to use this route (default: 0)
   */
  level?: number;

  /**
   * Optional link to the Zwift Insider page for this route.
   */
  link?: string;
  
  /**
   * Name of the route on Zwift.
   */
  name: string;

  /**
   * The sports that this route applies to.
   */
  sports?: string;

  /**
   * The Zwift World that holds this route.
   */
  world: string;
}


