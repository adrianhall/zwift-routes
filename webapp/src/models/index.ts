import capitalize from 'lodash/capitalize';

/**
 * Possible values for the display-units.  Either imperial (miles and feet),
 * or metric (kilometers and meters).
 */
export enum DisplayUnits {
  Imperial            = 'imperial',
  Metric              = 'metric'
}

/**
 * Possible values for the list sort direction - ascending or descending.
 */
export enum SortDirection {
  Ascending           = 'ascending',
  Descending          = 'descending'
}

/**
 * Possible values for the sort field.
 */
export enum SortField { 
  Name                = 'name',
  Difficulty          = 'difficulty',
  RouteDistance       = 'routeDistance',
  RouteElevationGain  = 'routeElevationGain',
  TotalDistance       = 'totalDistance',
  TotalElevationGain  = 'totalElevationGain'
}

/**
 * List of potential sports that can be done on Zwift.
 */
export enum ZwiftSport {
  AllSports           = '*',
  Cycling             = 'cycling',
  Running             = 'running'
}

/**
 * Definition of the display settings to use.
 */
export interface DisplaySettings {
  units: DisplayUnits;
}

/**
 * Definition of a filter used to limit the number of routes in the list.
 */
export interface RouteFilter {
  world?: string;
  sport?: ZwiftSport;
  includeCompletedRoutes: boolean;
  includeEventOnlyRoutes: boolean;
  includeDefaultWorld: boolean;
  maximumZwiftLevel?: number;
}

/**
 * When we sort, we sort by a field and direction.
 */
export interface SortOrder {
  sortField: SortField;
  sortDirection: SortDirection;
}

/**
 * The format of a single routes.json entry.
 */
export interface ZwiftRouteData {
  id: string;
  difficulty: number;
  distance: number;
  elevation: number;
  eventonly: boolean;
  leadin_distance: number;
  leadin_elevation: number;
  level?: number;
  link?: string;
  name: string;
  sports?: string;
  world: string;
}

/**
 * The format of the data written to / read from the IndexedDB
 */
export interface ZwiftUserData {
  routeId: string;
  isCompleted: boolean;
}

/**
 * A ZwiftRoute combines the static data from the routes.json file and
 * the dynamic data from the IndexedDB.
 */
export class ZwiftRoute {
  readonly id: string;
  
  readonly difficulty: number;
  isCompleted: boolean;
  readonly isEventOnly: boolean;
  readonly isForCycling: boolean;
  readonly isForRunning: boolean;
  readonly leadinDistance: number;
  readonly leadinElevationGain: number;
  readonly minimumZwiftLevel: number;
  readonly routeDistance: number;
  readonly routeElevationGain: number;
  readonly routeName: string;
  readonly totalDistance: number;
  readonly totalElevationGain: number;
  readonly zwiftInsiderLink?: URL;
  readonly zwiftWorld: string;

  constructor(staticData: ZwiftRouteData, dynamicData?: ZwiftUserData) {
    const sport = (staticData.sports || 'all').toLowerCase();

    this.id = staticData.id;
    this.difficulty = staticData.difficulty;
    this.isCompleted = dynamicData?.isCompleted || false;
    this.isEventOnly = staticData.eventonly;
    this.isForCycling = sport === 'all' || sport === 'cycling';
    this.isForRunning = sport === 'all' || sport === 'running';
    this.leadinDistance = staticData.leadin_distance;
    this.leadinElevationGain = staticData.leadin_elevation;
    this.minimumZwiftLevel = staticData.level || 0;
    this.routeDistance = staticData.distance;
    this.routeElevationGain = staticData.elevation;
    this.routeName = staticData.name;
    this.totalDistance = staticData.leadin_distance + staticData.distance;
    this.totalElevationGain = staticData.leadin_elevation + staticData.elevation;

    const w = staticData.world.split(/\s+/);
    this.zwiftWorld = w.map((word) => capitalize(word)).join(' ');

    if (staticData.link) {
      this.zwiftInsiderLink = new URL(staticData.link);
    }
  }

  /**
   * Gets the dynamic data for a route.
   */
  getZwiftUserData(): ZwiftUserData {
    return { routeId: this.id, isCompleted: this.isCompleted };
  }

  /**
   * Checks to see if a route needs to be included or excluded based on 
   * filter criteria.
   * 
   * @param filter the RouteFilter to use
   * @returns true if it is to be included; false if excluded.
   */
  isMatch(filter: RouteFilter): boolean {
    if (filter.world && filter.world !== '*') {
      const worldSet = [ filter.world ];
      if (filter.includeDefaultWorld) worldSet.push('Watopia');
      if (!worldSet.includes(this.zwiftWorld)) return false;
    }

    if (filter.sport && filter.sport !== ZwiftSport.AllSports) {
      if (filter.sport === ZwiftSport.Cycling && !this.isForCycling) return false;
      if (filter.sport === ZwiftSport.Running && !this.isForRunning) return false;
    }

    if (filter.maximumZwiftLevel) {
      if (this.minimumZwiftLevel >= filter.maximumZwiftLevel) return false;
    }

    if (!filter.includeCompletedRoutes && this.isCompleted) return false;
    if (!filter.includeEventOnlyRoutes && this.isEventOnly) return false;

    return true;
  }
}

