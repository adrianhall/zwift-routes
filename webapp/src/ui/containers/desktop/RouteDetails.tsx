import React from 'react';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import * as UI from 'src/ui/components'; 

import { DisplayUnits, ZwiftRoute } from 'src/models';
import ZwiftInsiderLink from 'src/ui/components/ZwiftInsiderLink';

const useStyles = makeStyles((theme: Theme) => createStyles({
  lower: {
    display: 'block'
  },
  upper: {
    display: 'flex',
    flexFlow: 'row nowrap',
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2)
  },
  badge: {
    display: 'block',
    padding: theme.spacing(2)
  },
  info: {
    flexGrow: 1,
    textAlign: 'center',
    paddingTop: theme.spacing(2)
  },
  links: {
    alignItems: 'center',
    display: 'flex',
    flexFlow: 'row nowrap'
  },
  paper: {
    padding: theme.spacing(2),
    paddingTop: theme.spacing(4)
  },
  zwiftInsiderLink: {
    flexGrow: 1
  },
  completedLink: {
    flexGrow: 1
  }
}));

export interface RouteDetailsProps {
  displayUnits?: DisplayUnits;
  onCompletedChanged?: (isCompleted: boolean) => void;
  route: ZwiftRoute;
}

const RouteDetails: React.SFC<RouteDetailsProps> = 
  ({ displayUnits = DisplayUnits.Metric, onCompletedChanged, route }: RouteDetailsProps) => {
    const styles = useStyles();

    const onCompleted = (): void => {
      if (onCompletedChanged) onCompletedChanged(!route.isCompleted);
    }

    return (
      <div className={styles.paper}>
        <Paper>
          <div className={styles.upper}>
            <div className={styles.badge}>
              <UI.WorldBadge world={route.zwiftWorld} />
            </div>
            <div className={styles.info}>
              <Typography variant="h4">{route.routeName}</Typography>
              <UI.ZwiftRouteIcons route={route} />
              <div className={styles.links}>
                <div className={styles.zwiftInsiderLink}>
                  <ZwiftInsiderLink route={route} />
                </div>
                <div className={styles.completedLink}>
                  <IconButton data-testid="completed-button" onClick={onCompleted}>
                    <UI.CompletedIcon isCompleted={route.isCompleted} />
                  </IconButton>
                </div>
              </div>
            </div>
          </div>
          <div className={styles.lower}>
            <UI.ZwiftRouteTable displayUnits={displayUnits} route={route} />
          </div>
        </Paper>
      </div>
    );
  };

export default RouteDetails;
