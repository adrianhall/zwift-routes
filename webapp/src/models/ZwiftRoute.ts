import capitalize from 'lodash/capitalize';
import ZwiftUserData from './ZwiftUserData';
import RouteFilter from './RouteFilter';
import ZwiftSport from './ZwiftSport';
import ZwiftRouteData from './ZwiftRouteData'

/**
 * The model for a single ZwiftRoute
 */
export default class ZwiftRoute {
  private staticData: ZwiftRouteData;
  private dynamicData: ZwiftUserData;
  private sport: string;

  constructor(staticData: ZwiftRouteData, dynamicData?: ZwiftUserData) {
    this.staticData = staticData;
    this.dynamicData = dynamicData || { routeId: this.staticData.id, isCompleted: false };
    this.sport = this.staticData.sports || 'all';
  }
  
  /**
   * Capitalize each word in a string.
   * 
   * @param s the input string
   */
  private capWords(s: string): string {
    const w = s.split(/\s+/);
    return w.map((word) => capitalize(word)).join(' ');
  }

  /**
   * Returns the unique ID for the route
   */
  get id(): string {
    return this.staticData.id;
  }

  /**
   * Returns the VeloViewer difficulty rating.
   */
  get difficulty(): number {
    return this.staticData.difficulty;
  }

  /**
   * Returns true if the user has completed the ride
   */
  get isCompleted(): boolean {
    return this.dynamicData.isCompleted;
  }

  /**
   * Sets or clears the user completion flag
   */
  set isCompleted(value: boolean) {
    this.dynamicData.isCompleted = value;
  }

  /**
   * Returns true if this route can only be used during events.
   */
  get isEventOnly(): boolean {
    return this.staticData.eventonly;
  }

  /**
   * Returns true if you can ride a bike on this route.
   */
  get isForCycling(): boolean {
    return this.sport === 'all' || this.sport === 'cycling';
  }

  /**
   * Returns true if you can run on this route.
   */
  get isForRunning(): boolean {
    return this.sport === 'all' || this.sport === 'running';
  }

  /**
   * Returns the distance from the drop-point to the route start (in km).
   */
  get leadinDistance(): number {
    return this.staticData.leadin_distance;
  }

  /**
   * Returns the elevation gain from the drop-point to the route start (in m).
   */
  get leadinElevationGain(): number {
    return this.staticData.leadin_elevation;
  }

  /**
   * The minimum level required to use this route.
   */
  get minimumZwiftLevel(): number {
    return this.staticData.level || 0;
  }

  /**
   * The official distance (start to end) of the route (in km).
   */
  get routeDistance(): number {
    return this.staticData.distance;
  }

  /**
   * The official elevation gain (start to end) of the route (in m).
   */
  get routeElevationGain(): number {
    return this.staticData.elevation;
  }

  /**
   * The official name of the route.
   */
  get routeName(): string {
    return this.staticData.name;
  }

  /**
   * The total distance (including lead-in) needed to complete this ride, in km.
   */
  get totalDistance(): number {
    return this.staticData.leadin_distance + this.staticData.distance;
  }

  /**
   * The total elevation gain (including lead-in) needed to complete 
   * this ride, in m.
   */
  get totalElevationGain(): number {
    return this.staticData.leadin_elevation + this.staticData.elevation;
  }

  /**
   * The URL to the Zwift Insider route description, or undefined if it
   * doesn't exist.
   */
  get zwiftInsiderLink(): string | undefined {
    return this.staticData.link;
  }

  /**
   * The name of the world.  We normalize the world name such that each
   * world name is capitalized, to allow for proper sorting and comparison.
   */
  get zwiftWorld(): string {
    return this.capWords(this.staticData.world);
  }

  /**
   * Returns the ZwiftUserData for this object.  This is used when storing
   * the data.
   */
  getZwiftUserData(): ZwiftUserData {
    return {
      routeId: this.id,
      isCompleted: this.isCompleted
    };
  }

  /**
   * Determines if this route is a match for the provided filter
   * 
   * @param filter the route filter to compare this route to
   * @returns true if this route matches the filter.
   */
  isMatch(filter: RouteFilter): boolean {
    // The general process is to escape out (return false) if something
    // doesn't match - if everything matches, then return true.
    if (filter.world) {
      const worldSet = [ filter.world ];
      if (filter.includeDefaultWorld) worldSet.push('Watopia');
      if (!worldSet.includes(this.zwiftWorld)) {
        return false;
      }
    }

    if (filter.sport && filter.sport === ZwiftSport.Cycling && !this.isForCycling) {
      return false;
    }

    if (filter.sport && filter.sport === ZwiftSport.Running && !this.isForRunning) {
      return false;
    }

    if (!filter.includeCompletedRoutes && this.isCompleted) {
      return false;
    }

    if (!filter.includeEventOnlyRoutes && this.isEventOnly) {
      return false;
    }

    if (filter.maximumZwiftLevel && this.minimumZwiftLevel >= filter.maximumZwiftLevel) {
      return false;
    }
    
    return true;
  }
}
