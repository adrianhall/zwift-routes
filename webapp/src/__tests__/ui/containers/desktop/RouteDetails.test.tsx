/* eslint-env jest */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import RouteDetails from 'src/ui/containers/desktop/RouteDetails';
import { ZwiftRoute, DisplayUnits } from 'src/models';
import routeData from 'src/data/routes.json';

describe('desktop', () => {
  describe('RouteDetails', () => {
    it('renders a basic route', () => {
      const route = new ZwiftRoute(routeData[0]);
      const { container } = render(<RouteDetails route={route} />);
      expect(container).toMatchSnapshot();
    });

    it('shows the world badge', () => {
      const route = new ZwiftRoute(routeData[0]);
      const { getByAltText } = render(<RouteDetails route={route} />);
      expect(getByAltText(route.zwiftWorld)).not.toBeNull();
    });

    it('shows the ZwiftInsiderLink', () => {
      const route = new ZwiftRoute(routeData[0]);
      const { queryByAltText } = render(<RouteDetails route={route} />);
      expect(queryByAltText("Zwift Insider")).not.toBeNull();
    });

    it('shows completed when complete', () => {
      const route = new ZwiftRoute(routeData[0]);
      const { queryByTestId } = render(<RouteDetails route={route} />);
      expect(queryByTestId('not-completed-icon')).not.toBeNull();
    });

    it('shows incomplete when incomplete', () => {
      const route = new ZwiftRoute(routeData[0], { routeId: routeData[0].id, isCompleted: true });
      const { queryByTestId } = render(<RouteDetails route={route} />);
      expect(queryByTestId('is-completed-icon')).not.toBeNull();
    });

    it('calls the callback when complete is clicked', () => {
      const callback = jest.fn();
      const route = new ZwiftRoute(routeData[0]);
      const { getByTestId } = render(<RouteDetails onCompletedChanged={callback} route={route} />);
      fireEvent.click(getByTestId('not-completed-icon'));
      expect(callback).toHaveBeenCalledTimes(1);
    });

    it('accepts clicks on completion indicator', () => {
      const route = new ZwiftRoute(routeData[0]);
      const { getByTestId, queryByTestId } = render(<RouteDetails route={route} />);
      expect(queryByTestId('not-completed-icon')).not.toBeNull();
      fireEvent.click(getByTestId('not-completed-icon'));
    });
  });
});
