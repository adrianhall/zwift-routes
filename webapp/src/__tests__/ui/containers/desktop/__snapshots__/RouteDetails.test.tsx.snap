// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`desktop RouteDetails renders a basic route 1`] = `
<div>
  <div
    class="makeStyles-paper-6"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="makeStyles-upper-2"
      >
        <div
          class="makeStyles-badge-3"
        >
          <div
            class="makeStyles-badge-37"
          >
            <img
              alt="Watopia"
              class="makeStyles-badge-37"
              src="Watopia.png"
            />
          </div>
        </div>
        <div
          class="makeStyles-info-4"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4"
          >
            The Pretzel
          </h4>
          <div
            class="makeStyles-iconContainer-68"
            style="font-size: 1.5rem;"
          >
            <div
              aria-label="Cycling"
              class="makeStyles-icon-69"
              title="Cycling"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                data-testid="cycling-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 5.5c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zM5 12c-2.8 0-5 2.2-5 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 8.5c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5 3.5 1.6 3.5 3.5-1.6 3.5-3.5 3.5zm5.8-10l2.4-2.4.8.8c1.3 1.3 3 2.1 5.1 2.1V9c-1.5 0-2.7-.6-3.6-1.5l-1.9-1.9c-.5-.4-1-.6-1.6-.6s-1.1.2-1.4.6L7.8 8.4c-.4.4-.6.9-.6 1.4 0 .6.2 1.1.6 1.4L11 14v5h2v-6.2l-2.2-2.3zM19 12c-2.8 0-5 2.2-5 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 8.5c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5 3.5 1.6 3.5 3.5-1.6 3.5-3.5 3.5z"
                />
              </svg>
            </div>
            <div
              aria-label="Running"
              class="makeStyles-icon-69"
              title="Running"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                data-testid="running-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13.49 5.48c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm-3.6 13.9l1-4.4 2.1 2v6h2v-7.5l-2.1-2 .6-3c1.3 1.5 3.3 2.5 5.5 2.5v-2c-1.9 0-3.5-1-4.3-2.4l-1-1.6c-.4-.6-1-1-1.7-1-.3 0-.5.1-.8.1l-5.2 2.2v4.7h2v-3.4l1.8-.7-1.6 8.1-4.9-1-.4 2 7 1.4z"
                />
              </svg>
            </div>
          </div>
          <div
            class="makeStyles-links-5"
          >
            <div
              class="makeStyles-zwiftInsiderLink-7"
            >
              <a
                href="http://zwiftinsider.com/the-pretzel/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Zwift Insider"
                  src="zwiftinsider.png"
                  style="height: 2rem; width: 2rem;"
                />
              </a>
            </div>
            <div
              class="makeStyles-completedLink-8"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                data-testid="completed-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    data-testid="not-completed-icon"
                    focusable="false"
                    role="presentation"
                    style="color: rgb(158, 158, 158);"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.59-13L12 10.59 8.41 7 7 8.41 10.59 12 7 15.59 8.41 17 12 13.41 15.59 17 17 15.59 13.41 12 17 8.41z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-lower-1"
      >
        <div
          class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            class="MuiTable-root makeStyles-table-103"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Â 
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                  scope="col"
                >
                  Distance
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                  scope="col"
                >
                  Elevation
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  Lead in
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  0.0km
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  0m
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  Route
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  72.4km
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  1360m
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  Total
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  72.4km
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  1360m
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  Difficulty
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                  colspan="2"
                >
                  34.53
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
